model/user.go

```diff
@@ -15,10 +15,13 @@ import (
 
 // User represents a user storage entity.
 type User struct {
-	Username       string
-	Password       string
-	LastPresence   *xmpp.Presence
-	LastPresenceAt time.Time
+	Username            string
+	PasswordScramSHA1   []byte
+	PasswordScramSHA256 []byte
+	Salt                []byte
+	IterationCount      int
```

Comment:
Perhaps it would be cleaner to have this is a separate Password struct

Reply:
I feel like the extra nesting would just make the code uglier, but I have no strong preference if the maintainer likes it one way or another.